webrecorder/browsertrix
# Add prettier for formatting frontend files

Adds js/css/html formatting with [Prettier](https://prettier.io/). Formatting hasn't been applied in this PR yet in case you have strong opinions about the default configuration.

### Testing
1. Install new dependencies with `yarn` and run `yarn format`. Files should be formatted in place
2. Create build `yarn start-dev`. Should be no errors

### Opinions
Prettier is opinionated. Defaults:
- Print width: 80
- Semicolons: true, adds semicolon to end of statements
- Quotes: double
- Trailing commas: added to objects and arrays
- Curly bracket spacing: added object literal brackets
- HTML bracket line: puts `>` in multi-line HTML in new line
- Array function parenthesis: always adds parens

The current code base follows similar rules for the most part. The most apparent difference is in the HTML, for example:
<img width="876" alt="Screen Shot 2021-11-18 at 11 30 25 AM" src="https://user-images.githubusercontent.com/4672952/142484077-e9dc2892-3572-445f-b735-8971c810064d.png">
